src/lib/utils.ts

```diff
@@ -53,3 +53,38 @@ export function formatAmount(
 		return amount;
 	}
 }
+
+/**
+ * Overloaded function to convert a string amount to token units
+ * with proper TS types for string or undefined.
+ */
+export function toTokenUnits(amount: string, decimals: number): string;
+export function toTokenUnits(amount: undefined, decimals: number): undefined;
+export function toTokenUnits(
+	amount: string | undefined,
+	decimals: number,
+): string | undefined {
+	if (amount === undefined) return undefined;
+	return floatToBigNumber(amount, decimals).toString();
+}
+
+/**
+ * Parse string to float, ensuring NaN -> 0
+ */
+export function parseAmountValue(amount: string): number {
+	const value = parseFloat(amount);
+	return Number.isNaN(value) ? 0 : value;
+}
+
+/**
+ * Generate a random 53-bit integer.
+ * (In JavaScript, we reliably handle up to 53 bits for integer precision.)
```

Comment:
This is why the type of `queyId` is usually a sting and not a number. 
To be able to use the whole 64 bits value